src/gcp/identityPlatform.ts

```diff
@@ -0,0 +1,234 @@
+  name?: string;
+  signIn?: {
+    email?: {
+      enabled: boolean;
+      passwordRequired: boolean;
+    };
+    phoneNumber?: {
+      enabled: boolean;
+      testPhoneNumbers: Record<string, string>;
+    };
+    anonymous?: {
+      enabled: boolean;
+    };
+    allowDuplicateEmails?: boolean;
+    hashConfig?: {
+      algorithm: HashAlgorithm;
+      signerKey: string;
+      saltSeparator: string;
+      rounds: number;
+      memoryCost: number;
+    };
+  };
+  notification?: {
+    sendEmail: {
+      method: "METHOD_UNSPECIFIED" | "DEFAULT" | "CUSTOM_SMTP";
+      resetPasswordTemplate: EmailTemplate;
+      verifyEmailTemplate: EmailTemplate;
+      changeEmailTemplate: EmailTemplate;
+      legacyResetPasswordTemplate: EmailTemplate;
+      callbackUri: string;
+      dnsInfo: {
+        customDomain: string;
+        useCustomDomain: boolean;
+        pendingCustomDomain: string;
+        customDomainState:
+          | "VERIFICATION_STATE_UNSPECIFIED"
+          | "NOT_STARTED"
+          | "IN_PROGRESS"
+          | "FAILED"
+          | "SUCCEEDED";
+        domainVerificationRequestTime: string;
+      };
+      revertSecondFactorAdditionTemplate: EmailTemplate;
+      smtp: {
+        senderEmail: string;
+        host: string;
+        port: number;
+        username: string;
+        password: string;
+        securityMode: "SECURITY_MODE_UNSPECIFIED" | "SSL" | "START_TLS";
+      };
+    };
+    sendSms: {
+      useDeviceLocale?: boolean;
+      smsTemplate?: {
+        content?: string;
+      };
+    };
+    defaultLocale?: string;
+  };
+  quota?: {
+    signUpQuotaConfig?: {
+      quota?: string;
+      startTime?: string;
+      quotaDuration?: string;
+    };
+  };
+  monitoring?: {
+    requestLogging?: {
+      enabled?: boolean;
+    };
+  };
+  multiTenant?: {
+    allowTenants?: boolean;
+    defaultTenantLocation?: string;
+  };
+  authorizedDomains?: Array<string>;
+  subtype?: "SUBTYPE_UNSPECIFIED" | "IDENTITY_PLATFORM" | "FIREBASE_AUTH";
+  client?: {
+    apiKey?: string;
+    permissions?: {
+      disabledUserSignup?: boolean;
+      disabledUserDeletion?: boolean;
+    };
+    firebaseSubdomain?: string;
+  };
+  mfa?: {
+    state?: "STATE_UNSPECIFIED" | "DISABLED" | "ENABLED" | "MANDATORY";
+    enabledProviders?: Array<Provider>;
+  };
+  blockingFunctions?: BlockingFunctionsConfig;
+}
+
+/**
+ * Helper function to get the blocking function config from identity platform.
+ * @param project GCP project ID or number
+ * @returns the blocking functions config
+ */
+export async function getBlockingFunctionsConfig(
+  project: string
+): Promise<BlockingFunctionsConfig> {
+  const config = (await getConfig(project)) || {};
+  if (!config.blockingFunctions) {
+    config.blockingFunctions = {
+      triggers: {
+        beforeCreate: {
+          functionUri: "",
+        },
+        beforeSignIn: {
+          functionUri: "",
+        },
+      },
+      forwardInboundCredentials: {},
```

Comment:
Per typing of `BlockingFunctionsConfig`, every property is optional. I think most our code assumes that these property may not exists. Why do we go about initializing an empty object?

Reply:
Good point, I'll remove it